v0/destinations/tiktok_ads/transform.js

```diff
@@ -0,0 +1,259 @@
+/* eslint-disable camelcase */
+const { SHA256 } = require("crypto-js");
+const get = require("get-value");
+const { EventType } = require("../../../constants");
+const {
+  getErrorRespEvents,
+  CustomError,
+  constructPayload,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultBatchRequestConfig,
+  getSuccessRespEvents,
+  isDefinedAndNotNullAndNotEmpty
+} = require("../../util");
+const {
+  trackMapping,
+  TRACK_ENDPOINT,
+  BATCH_ENDPOINT,
+  eventNameMapping,
+  MAX_BATCH_SIZE
+} = require("./config");
+
+function checkIfValidSHA256(str) {
+  // Regular expression to check if string is a SHA256 hash
+  const regexExp = /^[a-f0-9]{64}$/gi;
+
+  return regexExp.test(str);
+}
+
+function checkIfValidPhoneNumber(str) {
+  // Ref - https://ads.tiktok.com/marketing_api/docs?id=1701890979375106
+  // Regular expression to check whether it has country code
+  // but should not include +86
+  const regexExp = /^(\+(?!86)\d{1,3}){0,1}[0-9]{10}$/gi;
+
+  return regexExp.test(str);
+}
+
+const trackResponseBuilder = async (message, { Config }) => {
+  const pixel_code = Config.pixelCode;
+
+  let event = get(message, "event");
+  event = event ? event.trim().toLowerCase() : event;
+  if (!event) {
+    throw new CustomError("Event name is required", 400);
+  }
+
+  if (eventNameMapping[event] === undefined) {
+    throw new CustomError(`Event name (${event}) is not valid`, 400);
+  }
+  event = eventNameMapping[event];
+
+  let payload = constructPayload(message, trackMapping);
+  payload = { pixel_code, event, ...payload };
+
+  /*
+   * Hashing user related detail i.e external_id, email, phone_number
+   */
+
+  const external_id = payload.context.user.external_id.trim();
+  if (
+    isDefinedAndNotNullAndNotEmpty(external_id) &&
+    !checkIfValidSHA256(external_id)
+  ) {
+    payload.context.user.external_id = SHA256(external_id).toString();
+  }
+
+  const email = payload.context.user.email.trim().toLowerCase();
+  if (isDefinedAndNotNullAndNotEmpty(email) && !checkIfValidSHA256(email)) {
+    payload.context.user.email = SHA256(email).toString();
+  }
+
+  const phone_number = payload.context.user.phone_number.trim();
+  if (
+    isDefinedAndNotNullAndNotEmpty(phone_number) &&
+    !checkIfValidSHA256(phone_number)
+  ) {
+    if (checkIfValidPhoneNumber(phone_number)) {
+      payload.context.user.phone_number = SHA256(phone_number).toString();
+    } else {
+      throw new CustomError(
+        "Invalid phone number. Include proper country code except +86. Aborting ",
+        400
+      );
+    }
+  }
+
```

Comment:
Here without expecting the phone, email, and external_id always in SHA256 only, we can convert them to sha256 as well. Just provide a switch in config if the customer wants us to convert those attributes to SHA256 or they will provide in SHA256. WDYT @saikumarrs

Reply:
@akashrpo @ItsSudip I believe the default should be "non-hashed" as the same event data is flown to other destinations as well. Yeah, the config switch was already discussed to be supported.

Reply:
The support we discussed will be added in the case of device mode where it is optional. However, here the default should be "non-hashed" but we have a functionality here to accept it if given in the hashed form too.

Reply:
@akashrpo I wouldn't deduce the hashed status but drive it through configuration. There is a risk of actual data looking alike a hash.

Reply:
added a way to drive this from webapp configuration